keyVaultClientSideToCustomerManagedServerSide/migration.py

```diff
@@ -0,0 +1,84 @@
+import os
+from keyVaultClientSideToCustomerManagedServerSide.setup import config as cfg
+from azure.storage.blob import BlobServiceClient
+from azure.keyvault.keys import KeyClient
+from azure.identity import ClientSecretCredential
+from azure.keyvault.keys.crypto import CryptographyClient, EncryptionAlgorithm
+
+
+def get_key(k_client, k_name):
+    # get key from keyvault
+    encryption_key = k_client.get_key(k_name)
+
+    # return keyvault key
+    return encryption_key
+
+
+def get_blob(blob_service_client, cont_name, blob_name):
+    # get blob contents and return in bytes
+    print("Reading blob from Azure Storage...\n")
+    # access specific container and blob using blob client that you want to download from
+    blob_client = blob_service_client.get_blob_client(container=cont_name, blob=blob_name)
+    # write contents from blob into a file as bytes and download
+    blob_content = blob_client.download_blob().readall()
+
+    # return contents from file in bytes
+    return blob_content
+
+
+def decrypt_blob(encryption_key, my_credential, encrypted_content, blob_name):
+    # decrypt blob-- requires data in form of bytes
+    print("Decrypting blob...")
+    # use keyvault key and credential to access account client for decryption
+    crypto_client = CryptographyClient(encryption_key, credential=my_credential)
+    # decrypt content
+    decrypted = crypto_client.decrypt(EncryptionAlgorithm.rsa_oaep, encrypted_content)
+    # convert decrypted content to plaintext to make it readable
+    decrypted_message = decrypted.plaintext
```

Comment:
Is this right? Are we supposed to
1. Download the blob
2. Pass the contents of the blob to KeyVault SDK (CryptogrpahyClient) and this decrypts for us??
3. then we get our decrypted stuff

I thought our blob client when we download was supposed to decrypt it for us

Reply:
I think you are right Amanda! 

---
In reply to: [454005332](https://github.com/wastore/azure-storage-samples-for-python/pull/4#discussion_r454005332) [](ancestors = 454005332)